src/batched/KokkosBatched_Gemm_Decl.hpp

```diff
@@ -95,118 +146,125 @@ struct BatchedSerialGemm;
 ///                        matrix C is BxMxN
 /// \return 0 upon success, non-zero otherwise
-template <typename ArgTransA, typename ArgTransB, typename ArgBatchLayout>
-struct BatchedGemm {
-  template <typename ScalarType, typename AViewType, typename BViewType,
-            typename CViewType>
-  static int invoke(const ScalarType alpha, const AViewType &A,
-                    const BViewType &B, const ScalarType beta,
-                    const CViewType &C) {
-    int ret = 0;
-    // Check for valid input views
-    static_assert(Kokkos::Impl::is_view<AViewType>::value,
-                  "AViewType must be a Kokkos::View.");
-    static_assert(Kokkos::Impl::is_view<BViewType>::value,
-                  "BViewType must be a Kokkos::View.");
-    static_assert(Kokkos::Impl::is_view<CViewType>::value,
-                  "CViewType must be a Kokkos::View.");
-    static_assert(static_cast<int>(AViewType::rank) == 3,
-                  "AViewType must have rank 3.");
-    static_assert(static_cast<int>(BViewType::rank) == 3,
-                  "BViewType must have rank 3.");
-    static_assert(static_cast<int>(CViewType::rank) == 3,
-                  "CViewType must have rank 3.");
-
-    // Check for valid data access patterns
-    // Skip checking a_layout == b_layout == c_layout
-    // Skip checking for LayoutStride
-    using c_layout = typename CViewType::array_layout;
-    if (std::is_same<c_layout, Kokkos::LayoutLeft>::value &&
-        !std::is_same<ArgBatchLayout, BatchLayout::Right>::value) {
-      throw std::runtime_error(
-          "Error: LayoutLeft views require BatchLayout::Right");
-    }
-    if (std::is_same<c_layout, Kokkos::LayoutRight>::value &&
-        !std::is_same<ArgBatchLayout, BatchLayout::Left>::value) {
-      throw std::runtime_error(
-          "Error: LayoutRight views require BatchLayout::Left");
-    }
+template <typename ArgTransA, typename ArgTransB, typename ArgBatchSzDim,
+          typename BatchedGemmHandleType, typename ScalarType,
+          typename AViewType, typename BViewType, typename CViewType>
+int BatchedGemm(const BatchedGemmHandleType *handle, const ScalarType alpha,
+                const AViewType &A, const BViewType &B, const ScalarType beta,
+                const CViewType &C) {
+  int ret = 0;
+  // Check for valid input views
+  static_assert(Kokkos::Impl::is_view<AViewType>::value,
+                "AViewType must be a Kokkos::View.");
+  static_assert(Kokkos::Impl::is_view<BViewType>::value,
+                "BViewType must be a Kokkos::View.");
+  static_assert(Kokkos::Impl::is_view<CViewType>::value,
+                "CViewType must be a Kokkos::View.");
+  static_assert(static_cast<int>(AViewType::rank) == 3,
+                "AViewType must have rank 3.");
+  static_assert(static_cast<int>(BViewType::rank) == 3,
+                "BViewType must have rank 3.");
+  static_assert(static_cast<int>(CViewType::rank) == 3,
+                "CViewType must have rank 3.");
+
```

Comment:
In the description above you state that `{A,B,C}ViewType::rank` can be 3 or 4 but here you require it to be 3?
Is the SIMD path not working yet or am I missing something?